src/nest.cpp

```diff
@@ -0,0 +1,24 @@
+#include <Rcpp.h>
+#include <dplyr.h>
+using namespace Rcpp;
+
+// [[Rcpp::export]]
+SEXP extract_groups(const List data, const CharacterVector nested_cols) {
+  ListOf<IntegerVector> indices(data.attr("indices"));
```

Comment:
ow, I didn't know that. I always assumed indices were kept in a working state.

I don't see that behaviour with `filter()` but it may be safer to use your original implementation.